src/main/scala/com/microsoft/partnercatalyst/fortis/spark/dba/CassandraConfigurationManager.scala

```diff
@@ -0,0 +1,88 @@
+package com.microsoft.partnercatalyst.fortis.spark.dba
+import java.util.concurrent.ConcurrentHashMap
+
+import com.microsoft.partnercatalyst.fortis.spark.dto.{BlacklistedTerm, Geofence, SiteSettings}
+import com.microsoft.partnercatalyst.fortis.spark.sources.streamprovider.ConnectorConfig
+import org.apache.spark.SparkContext
+import com.datastax.spark.connector._
+import com.microsoft.partnercatalyst.fortis.spark.logging.Loggable
+import scala.compat.java8.FunctionConverters._
+
+@SerialVersionUID(100L)
+class CassandraConfigurationManager extends ConfigurationManager with Serializable with Loggable {
+  // Note: trusted sources are cached for the lifetime of the configuration manager since in order to update them,
+  // streaming must be restarted (and hence the configuration manager would be replaced).
+  private lazy val connectorToTrustedSources = new ConcurrentHashMap[String, Seq[String]]()
+
+  override def fetchConnectorConfigs(sparkContext: SparkContext, pipeline: String): List[ConnectorConfig] = {
+    def fetchTrustedSources(connectorName: String): Seq[String] = {
+      sparkContext.cassandraTable(CassandraSchema.KeyspaceName, CassandraSchema.Table.TrustedSourcesName)
+        .select("externalsourceid")
+        .map(row => row.getString("externalsourceid")).collect()
+    }
+
+    val pipelineConfigRows = sparkContext.cassandraTable[CassandraSchema.Table.Stream](CassandraSchema.KeyspaceName,
+      CassandraSchema.Table.StreamsName).collect()
```

Comment:
Hey so what I'm doing here is bringing the stream row objects from the worker nodes to the driver. The reason for this is so that the map call below it will execute on the driver (I do more Cassandra queries in there, and also access a local cache on the driver).